src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -857,6 +891,36 @@ static Keyspaces fetchNonSystemKeyspaces()
         return fetchKeyspacesWithout(SchemaConstants.LOCAL_SYSTEM_KEYSPACE_NAMES);
     }
 
+    public static void validateNonCompact() throws StartupException
+    {
+        String query = format("SELECT keyspace_name, table_name, flags FROM %s.%s", SchemaConstants.SCHEMA_KEYSPACE_NAME, TABLES);
+
+        String messages = "";
+        for (UntypedResultSet.Row row : query(query))
+        {
+            if (SchemaConstants.isLocalSystemKeyspace(row.getString("keyspace_name")))
+                continue;
+
+            Set<String> flags = row.getSet("flags", AsciiType.instance);
```

Comment:
This is probably going to work given the possible values of the `flags` column, but I think that the proper call for the definition of the `flags` column on `system_schema.tables` would be:
```suggestion
            Set<String> flags = row.getFrozenSet("flags", UTF8Type.instance);
```